### PR TITLE
[1pt] PR: inundate_nation.py for FIM4

### PR DESCRIPTION
**Pull request by CalebOliven-NOAA**
This pull request is to merge `inundate_nation.py` into the dev code base. This pr changes the file `inundate_nation.py` to no longer call to the function `inundate.py`, but instead use `inundate_gms.py` and `mosaic_inundation.py`. This will make the file `inundate_nation.py` compatible with the FIM4 (gms) workflow.

## Additions

-`inundate_gms.py` Only one change was needed in the file `inundate_gms.py`. This was on line 214 where the `mask_type` has been changed to 'filter'. This change is to accommodate the fix to allow inundation across huc8 boundaries.
-`inundate_nation.py` Several changes were needed in this file to allow for correct interaction with the new calls to `inundate_gms.py` and `mosaic_inundation.py`.

## Testing

1. To run a test without, the command `python inundate_nation.py -r "/data/outputs/fim_4_0_5_x_sample_9hucs/" -o "{your output path here}" -j 4 -s`

## Screenshots
-Here is a sample output of mosaiced inundation rasters as a single `.tif`
image

## Todos

-Todo in the future is to add depth raster production capabilities.

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
+## v4.0.6.1 - 2022-08-11 - [PR #654](https://github.com/NOAA-OWP/inundation-mapping/pull/654)
+
+`inundate_nation.py` A change to switch the inundate nation function away from refrences to `inundate.py`, and rather use `inundate_gms.py` and `mosaic_inundation.py`
+## Changes
+
+- `inundate_gms`:  Changed line 214 to use `mask_type = 'filter'`
+
+<br/><br/>
+
+=======
```
Comment (CarsonPruitt-NOAA) [inline docs/CHANGELOG.md:13]: These = are special characters and cause issues. Let's take them out. Also, let's make sure that the changelog Changes and Additions sections match your PR.

**Review by RyanSpies-NOAA — DISMISSED** [review]
I successfully tested this with two different sets of HUCs as well as multiple recurrence intervals. Note that the default output & temp directories are probably not synced with our production machine, so there may be some set up work needed to run this for the full BED domain on production.

**Comment by CarsonPruitt-NOAA** [comment]
@CalebOliven-NOAA Go ahead and update the CHANGELOG on this and I'll merge it in. This will be v4.0.7.2. Thanks!

**Comment by CalebOliven-NOAA** [comment]
Change log updated after pull. Pulled from fim-nation.

**Review by CarsonPruitt-NOAA — APPROVED** [review]

**Comment by CalebOliven-NOAA** [comment]
Sure thing, Thanks for reviewing it.

Caleb

**Pull request merged**